scripts/dts/extract/globals.py

```diff
@@ -287,21 +287,21 @@ def add_prop_aliases(node_path,
 def get_binding(node_path):
     compat = get_compat(node_path)
 
-    # For just look for the binding in the main dict
-    # if we find it here, return it, otherwise it best
-    # be in the bus specific dict
-    if compat in bindings:
-        return bindings[compat]
-
+    # First look for a bus-specific binding. If not found, look in the
+    # main dict.
     parent_path = get_parent_path(node_path)
     parent_compat = get_compat(parent_path)
 
-    parent_binding = bindings[parent_compat]
+    try:
+        parent_binding = bindings[parent_compat]
+        bus = parent_binding['child']['bus']
+        binding = bus_bindings[bus][compat]
+        return binding
+    except (KeyError, TypeError):
```

Comment:
When do you get `TypeError`? If it's deliberate, then a comment might be worthwhile.

IMO, it might be clearer to explicitly test for membership (`if parent_compat in bindings: ...`).

Reply:
@ulfalizer Thanks. I've implemented another approach based on your feedback. Please review.